instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/servlet/AppServerBridge.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.servlet;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Helper containers for the Context attributes for transferring certain information between servlet
+ * integration and app-server server handler integrations.
+ */
+public class AppServerBridge {
+
+  public static final ContextKey<AtomicReference<Throwable>> THROWABLE_CONTEXT_KEY =
+      ContextKey.named("opentelemetry-servlet-unhandled-throwable");
+  public static final ContextKey<AtomicBoolean> SERVLET_SUGGESTED_BETTER_NAME_KEY =
+      ContextKey.named("opentelemetry-servlet-better-name-suggested");
+
+  public static Context init(Context ctx) {
+    return ctx.with(AppServerBridge.THROWABLE_CONTEXT_KEY, new AtomicReference<>())
+        .with(AppServerBridge.SERVLET_SUGGESTED_BETTER_NAME_KEY, new AtomicBoolean(false));
+  }
+
+  /**
+   * This is for servlet instrumentation to record exceptions unhandled by servlets and filters,
+   * that app-server integrations are interested in (to fail the span exceptionally with attached
+   * throwable), but which may be swallowed by an app-server before execution arrives to an
```

Comment:
Why exactly do we want to record exception in SERVER span if that was swallowed inside it?

Reply:
I changed this part as you have suggested earlier - record throwable from the outermost servlet/filter instead of storing it in the context. Will push soon.

Reply:
> I don't like `AppServerBridge` at all. I have no better suggestions right now, but I will think about it...

After removing throwables from this bridge, the only thing it has left is the flag for the updated name. The flag is only used by the servlet instrumentation, but it must have the same lifetime as the context created by the app server, so this bridge must be added to the context by the server instrumentation. You are welcome to suggest a better name or  more elegant approach altogether.